bpfd/src/multiprog/tc.rs

```diff
@@ -294,31 +288,21 @@ impl TcDispatcher {
         let path = format!("{base}/dispatcher_{}_{}", self.if_index, self.revision);
         fs::remove_dir_all(path)
             .map_err(|e| BpfdError::Error(format!("unable to cleanup state: {e}")))?;
-        // FIXME: Dispatcher *SHOULD* be detached when this object is dropped
-        if let Some(link) = self.link.take() {
-            link.detach()?;
-        }
+        if let Some(old_handle) = self.handle {
+            let attach_type = match self.direction {
+                Direction::Ingress => TcAttachType::Ingress,
+                Direction::Egress => TcAttachType::Egress,
+            };
+            if let Ok(old_link) =
+                SchedClassifierLink::attached(&self.if_name, attach_type, self.priority, old_handle)
+            {
+                old_link.detach()?;
```

Comment:
Does the `old_link` var get created from `if let Ok(as)`?

Reply:
I'm not sure what you mean by `if let Ok(as)`.  `old_link` gets re-instantiated with the `SchedClassifierLink::attached()`.  `attached()` is effectively a `new()` call, but Alessandro thought calling it `new()` would be confusing.

Reply:
Ahh cool makes sense I was confused because my cod diff was rendering incorrectly